Build broken on Mac OS (maybe more?) due to usage of removed std::random_shuffle in C++17

**Describe the bug**

pVst.cpp calls `std::random_shuffle(d->total.begin(), d->total.end());`

random_shuffle was deprecated in C++14 and fully removed in C++17: https://en.cppreference.com/w/cpp/algorithm/random_shuffle

It looks like Apple Clang is more pedantic than whatever compiler vcflib is built and distributed with on Linux, because when built on Mac OS with Apple Clang, building with C++17 fails. This will fail with any compiler that enforces the c++17 spec.

It looks like compiling with C++11 doesn't work because of the usage of some newer features like `auto` in lambdas, but building with C++14 should work. There's also the option to follow the recommended migration path: https://stackoverflow.com/questions/45013977/random-shuffle-is-not-a-member-of-std-error

**To Reproduce**

Compile with Apple Clang:
```
git clone --recursive https://github.com/vcflib/vcflib.git
cd vcflib
mkdir -p build && cd build
cmake -DCMAKE_BUILD_TYPE=Debug -DZIG=OFF -DOPENMP=OFF ..
cmake --build .
cmake --install .
```

Fail:
```
/Users/alex/src/vcflib/src/pVst.cpp:191:10: fatal error: no member named 'random_shuffle' in namespace 'std'
    std::random_shuffle(d->total.begin(), d->total.end());
    ~~~~~^
```
**Expected behavior**

vcflib should build.

--- a/src/pVst.cpp
+++ b/src/pVst.cpp
@@ -24,6 +24,7 @@
 #include <algorithm>
 #include <ctime>
 #include <cstdlib>
+#include <random>
 #include "gpatInfo.hpp"
 
 #if defined HAS_OPENMP
@@ -188,7 +189,9 @@ void calc(copyNcounts * d){
 
     trials += 1;
 
-    std::random_shuffle(d->total.begin(), d->total.end());
+    static std::mt19937 g{std::random_device{}()};
+
+    std::shuffle(d->total.begin(), d->total.end(), g);
 
     int tsize = d->target.size();
 

